interceptor/tracing_interceptor.go

```diff
@@ -357,7 +368,20 @@ func (t *tracingActivityInboundInterceptor) ExecuteActivity(
 
 type tracingWorkflowInboundInterceptor struct {
 	WorkflowInboundInterceptorBase
-	root *tracingInterceptor
+	root        *tracingInterceptor
+	spanCounter uint16
+	info        *workflow.Info
+}
+
+// newIdempotencyKey returns a new idempotency key by incrementing the span counter and interpolating
+// this new value into a string that includes the workflow namespace/id/run id and the interceptor type.
+func (t *tracingWorkflowInboundInterceptor) newIdempotencyKey() string {
+	t.spanCounter++
+	return fmt.Sprintf("WorkflowInboundInterceptor:%s:%s:%s:%d",
+		t.info.Namespace,
+		t.info.WorkflowExecution.ID,
+		t.info.WorkflowExecution.RunID,
```

Comment:
Should be noted that technically this value changes if a workflow is reset to a previous task. However, not only do we not appear to provide the "original run ID", but I think this is probably the correct behavior anyways.

Reply:
Yep I like to think of workflow reset as a fancy continue-as-new. No problem with having a new trace for a new workflow run.

Reply:
You can end up with orphaned spans this way if a wrapping workflow span crosses a reset point